src/test/java/nova/core/util/ProgressBarTest.java

```diff
@@ -40,30 +46,83 @@
 	@Rule
 	public ExpectedException thrown = ExpectedException.none();
 
-	EventBus<Event> events;
-
-    @Before
-    public void setUp() {
-		this.events = new EventBus<>();
-		this.events.on(ProgressBarEvent.class).bind(evt -> System.out.println(evt.message));
-	}
-
 	@Test
 	public void testProgressBar() {
+		EventBus<Event> events = new EventBus<>();
+		List<Object[]> testContents = Arrays.asList(
+			new Object[]{"Progress bar testing generic String message"},
+			new Object[]{"Progress bar testing class message", ProgressBarTest.class},
+			new Object[]{ProgressBarTest.class, "Progress bar testing class message"},
+			new Object[]{ProgressBarTest.class},
+			new Object[]{"Progress bar testing mod message", TestMod.class},
+			new Object[]{TestMod.class, "Progress bar testing mod message"},
+			new Object[]{TestMod.class},
+			new Object[]{"Progress bar testing null message", null},
+			new Object[]{null, "Progress bar testing class message"},
+			new Object[]{(String) null},
+			new Object[]{(Class) null},
+			new Object[]{ProgressBarTest.class, null},
+			new Object[]{null, ProgressBarTest.class});
+		List<String> testResults = new LinkedList<>();
+		events.on(TestProgressBarEvent.class).bind(evt -> testResults.add(evt.message));
 		ProgressBar progressBar = new AbstractProgressBar() {
 			@Override
 			protected void stepImpl(String message) {
-				events.publish(new ProgressBarEvent(message));
+				events.publish(new TestProgressBarEvent(message));
 			}
 		};
-		progressBar.step("Progress bar testing generic String message");
-		progressBar.step("Progress bar testing class message", ProgressBarTest.class);
-		progressBar.step(ProgressBarTest.class, "Progress bar testing class message");
-		progressBar.step(ProgressBarTest.class);
-		progressBar.step("Progress bar testing mod message", TestMod.class);
-		progressBar.step(TestMod.class, "Progress bar testing mod message");
-		progressBar.step(TestMod.class);
+		for (Object[] testArg : testContents) {
+			step(progressBar, testArg);
+		}
 		progressBar.finish();
+
+		List<String> expectedResults = testContents.stream().map(ProgressBarTest::convert).collect(Collectors.toList());
+
+		assertThat(testResults.size()).isEqualTo(testContents.size());
+		assertThat(testResults.size()).isEqualTo(expectedResults.size());
+
+		for (int i = 0; i < testContents.size(); i++)
+			assertThat(testResults.get(i)).isEqualTo(expectedResults.get(i));
```

Comment:
This can be done better using assertj collection assertions. For example 

```java
assertThat(foo).containsExactly(
    "Progress bar testing generic String message",
    "Progress bar testing class message: ProgressBarTest"
);
```